In [2]:
import pandas as pd

# 1 Traitement des dataframes et nettoyage des données

In [6]:
data_2019 = pd.read_csv('../raw_data/dcc-2019-pedestrian-footfall-count-jan-dec_14082020.csv')

In [7]:
data_2019

Time  O'Connell St Outside Pennys  \
0      01-01-2019 00:00:00                          244   
1      01-01-2019 00:15:00                          454   
2      01-01-2019 00:30:00                          391   
3      01-01-2019 00:45:00                          415   
4      01-01-2019 01:00:00                          319   
...                    ...                          ...   
35031  31-12-2019 22:45:00                          389   
35032  31-12-2019 23:00:00                          324   
35033  31-12-2019 23:15:00                          306   
35034  31-12-2019 23:30:00                          249   
35035  31-12-2019 23:45:00                          302   

       O'Connell St Outside Clerys  Mary Street  Capel Street  Aston Quay  \
0                           1914.0           20            16         NaN   
1                              NaN           24            62         NaN   
2                              NaN           69            99         NaN   
3                              NaN           50            61         NaN   
4                            885.0           24            47         NaN   
...                            ...          ...           ...         ...   
35031                          NaN           13            38       797.0   
35032                       1967.0           18            39       914.0   
35033                          NaN           19            69       724.0   
35034                          NaN           11            69       971.0   
35035                          NaN           14            52       513.0   

       Grafton Street @ CompuB  Talbot Street North  \
0                          NaN                  NaN   
1                          NaN                  NaN   
2                          NaN                  NaN   
3                          NaN                  NaN   
4                          NaN                  NaN   
...                        ...                  ...   
35031                    782.0                 52.0   
35032                    743.0                 48.0   
35033                    821.0                 38.0   
35034                    795.0                 69.0   
35035                    556.0                 47.0   

       Doilier Street, Burgh Quay  Dawson Street Replacement  ...  \
0                             NaN                        NaN  ...   
1                             NaN                        NaN  ...   
2                             NaN                        NaN  ...   
3                             NaN                        NaN  ...   
4                             NaN                        NaN  ...   
...                           ...                        ...  ...   
35031                       173.0                        6.0  ...   
35032                       220.0                        1.0  ...   
35033                       241.0                        4.0  ...   
35034                       357.0                        5.0  ...   
35035                       339.0                        1.0  ...   

       College Green, Bank Of Ireland  Henry Street  Westmoreland Street East  \
0                                 NaN          95.0                    1670.0   
1                                 NaN         265.0                       NaN   
2                                 NaN         117.0                       NaN   
3                                 NaN         120.0                       NaN   
4                                 NaN          90.0                     767.0   
...                               ...           ...                       ...   
35031                           283.0           NaN                       NaN   
35032                           287.0           NaN                    1217.0   
35033                           266.0           NaN                       NaN   
35034                           302.0           NaN                       NaN   
35035  

In [75]:
data_2019.shape

(35036, 24)

In [8]:
data_2019.isna().sum()

Time                                                   0
O'Connell St Outside Pennys                            0
O'Connell St Outside Clerys                        26277
Mary Street                                            0
Capel Street                                           0
Aston Quay                                          5664
Grafton Street @ CompuB                             5472
Talbot Street North                                 5641
Doilier Street, Burgh Quay                          5484
Dawson Street Replacement                          31673
Dame Street                                         5472
Talbot Street South                                 5496
O'Connell St, Parnell St @ AIB                      7400
Grafton Street / Nassau Street / Suffolk Street     5472
College Green, Bank Of Ireland                      5472
Henry Street                                        6624
Westmoreland Street East                           26277
Dawson Street                  

##  1-1 Je splite le dataframe 2019 pour traiter les données horaires uniquement.

In [127]:
data_2019_cleaned = data_2019.loc[:,['Time','O\'Connell St Outside Clerys', 'Dawson Street Replacement', 'Westmoreland Street East', 'Dawson Street',\
'Liffey Street', 'Westmoreland Street West', 'Grafton Street']]

In [128]:
data_2019_cleaned

Time  O'Connell St Outside Clerys  \
0      01-01-2019 00:00:00                       1914.0   
1      01-01-2019 00:15:00                          NaN   
2      01-01-2019 00:30:00                          NaN   
3      01-01-2019 00:45:00                          NaN   
4      01-01-2019 01:00:00                        885.0   
...                    ...                          ...   
35031  31-12-2019 22:45:00                          NaN   
35032  31-12-2019 23:00:00                       1967.0   
35033  31-12-2019 23:15:00                          NaN   
35034  31-12-2019 23:30:00                          NaN   
35035  31-12-2019 23:45:00                          NaN   

       Dawson Street Replacement  Westmoreland Street East  Dawson Street  \
0                            NaN                    1670.0          255.0   
1                            NaN                       NaN            NaN   
2                            NaN                       NaN            NaN   
3                            NaN                       NaN            NaN   
4                            NaN                     767.0          209.0   
...                          ...                       ...            ...   
35031                        6.0                       NaN            NaN   
35032                        1.0                    1217.0            NaN   
35033                        4.0                       NaN            NaN   
35034                        5.0                       NaN            NaN   
35035                        1.0                       NaN            NaN   

       Liffey Street  Westmoreland Street West  Grafton Street  
0              881.0                    1988.0           140.0  
1                NaN                       NaN             NaN  
2                NaN                       NaN             NaN  
3                NaN                       NaN             NaN  
4              723.0                    1270.0           215.0  
...              ...                       ...             ...  
35031            NaN                       NaN             NaN  
35032          936.0                    2898.0          1220.0  
35033            NaN                       NaN             NaN  
35034            NaN                       NaN             NaN  
35035            NaN                       NaN             NaN  

[35036 rows x 8 columns]

In [130]:
data_2019_cleaned['Dawson total']=\
pd.concat([data_2019_cleaned['Dawson Street'].iloc[:30000],\
           data_2019_cleaned['Dawson Street Replacement'].iloc[30000:]])

In [131]:
data_2019_cleaned = data_2019_cleaned.drop(columns=['Dawson Street','Dawson Street Replacement'])

In [132]:
data_2019_cleaned.shape[0]

35036

In [133]:
for i in range(data_2019_cleaned.shape[0]):
    if i%4 != 0 :
        data_2019_cleaned = data_2019_cleaned.drop(index = i)

In [134]:
data_2019_cleaned

Time  O'Connell St Outside Clerys  \
0      01-01-2019 00:00:00                       1914.0   
4      01-01-2019 01:00:00                        885.0   
8      01-01-2019 02:00:00                        984.0   
12     01-01-2019 03:00:00                        935.0   
16     01-01-2019 04:00:00                        390.0   
...                    ...                          ...   
35016  31-12-2019 19:00:00                       1750.0   
35020  31-12-2019 20:00:00                       1644.0   
35024  31-12-2019 21:00:00                       1554.0   
35028  31-12-2019 22:00:00                       1972.0   
35032  31-12-2019 23:00:00                       1967.0   

       Westmoreland Street East  Liffey Street  Westmoreland Street West  \
0                        1670.0          881.0                    1988.0   
4                         767.0          723.0                    1270.0   
8                         642.0         1010.0                    1589.0   
12                        582.0          772.0                    1534.0   
16                        143.0          197.0                     610.0   
...                         ...            ...                       ...   
35016                     763.0          910.0                    1929.0   
35020                     727.0          989.0                    1984.0   
35024                     752.0          779.0                    2120.0   
35028                     997.0          823.0                    2461.0   
35032                    1217.0          936.0                    2898.0   

       Grafton Street  Dawson total  
0               140.0         255.0  
4               215.0         209.0  
8               210.0         193.0  
12              204.0         178.0  
16               88.0          35.0  
...               ...           ...  
35016          1190.0          43.0  
35020          1239.0          61.0  
35024          1314.0           6.0  
35028          1315.0           9.0  
35032          1220.0           1.0  

[8759 rows x 7 columns]

In [135]:
data_2019_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8759 entries, 0 to 35032
Data columns (total 7 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Time                         8759 non-null   object 
 1   O'Connell St Outside Clerys  8759 non-null   float64
 2   Westmoreland Street East     8759 non-null   float64
 3   Liffey Street                8759 non-null   float64
 4   Westmoreland Street West     8759 non-null   float64
 5   Grafton Street               8759 non-null   float64
 6   Dawson total                 7607 non-null   float64
dtypes: float64(6), object(1)
memory usage: 547.4+ KB


pour Dawson Street, le comptage s'arrête à l'index 27064    173.0.

C'est environ le 281ème jour.

Pour Dawson Street Replacement, le comptage commence à l'index 31673  2157.0

C'est le 35ème jour avant le 31/12, ce qui correspond au 27/11 indiqué.

In [80]:
data_2020 = pd.read_csv('../raw_data/jan-dec-2020-ped-data.csv')

In [81]:
data_2020

Date & Time  Aston Quay  Bachelors Walk  \
0     01-01-2020 00:00:00      1600.0           868.0   
1     01-01-2020 01:00:00       858.0           727.0   
2     01-01-2020 02:00:00      3499.0           930.0   
3     01-01-2020 03:00:00      1804.0           561.0   
4     01-01-2020 04:00:00      2562.0           269.0   
...                   ...         ...             ...   
8779  31-12-2020 19:00:00        11.0           204.0   
8780  31-12-2020 20:00:00        30.0            91.0   
8781  31-12-2020 21:00:00       139.0            85.0   
8782  31-12-2020 22:00:00       162.0            44.0   
8783  31-12-2020 23:00:00        86.0            97.0   

      Capel Street - Removed from site 20/10  College Green - Dame St Side  \
0                                      277.0                        1797.0   
1                                      214.0                         594.0   
2                                      219.0                         429.0   
3                                      176.0                         365.0   
4                                       98.0                         176.0   
...                                      ...                           ...   
8779                                     NaN                          97.0   
8780                                     NaN                          59.0   
8781                                     NaN                          48.0   
8782                                     NaN                          35.0   
8783                                     NaN                          22.0   

      College Green @ Church Lane  College Green, Bank Of Ireland  \
0                          1494.0                           893.0   
1                           676.0                          1091.0   
2                           523.0                          1202.0   
3                           520.0                          1115.0   
4                           238.0                          1140.0   
...                           ...                             ...   
8779                        123.0                           234.0   
8780                         64.0                           153.0   
8781                         71.0                           164.0   
8782                         62.0                           139.0   
8783                         44.0                           147.0   

      Dame Street (Counter Missing)  Dawson Street Replacement  \
0                            1090.0                       18.0   
1                            1148.0                        7.0   
2                            1052.0                       15.0   
3                             899.0                        8.0   
4                             373.0                        9.0   
...                             ...                        ...   
8779                            NaN                      121.0   
8780                            NaN                       81.0   
8781                            NaN                      129.0   
8782                            NaN                      120.0   
8783                            NaN                       67.0   

      Doilier Street, Burgh Quay  ...  Henry Street  Liffey Street  \
0                          994.0  ...           NaN         1180.0   
1                          745.0  ...           NaN         1133.0   
2                          710.0  ...           NaN          898.0   
3                         1065.0  ...           NaN          595.0   
4                          768.0  ...           NaN          219.0   
...                          ...  ...           ...            ...   
8779                      1399.0  ...        1013.0          194.0   
8780                      1515.0  ...        1377.0          153.0   
8781                      1285.0  ...         271.0           86.0   
8782                       886.0  ...         213.0           66.0   


In [92]:
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8784 entries, 0 to 8783
Data columns (total 23 columns):
 #   Column                                           Non-Null Count  Dtype  
---  ------                                           --------------  -----  
 0   Aston Quay                                       8783 non-null   float64
 1   Bachelors Walk                                   8684 non-null   float64
 2   Capel Street - Removed from site 20/10           6791 non-null   float64
 3   College Green - Dame St Side                     8783 non-null   float64
 4   College Green @ Church Lane                      8783 non-null   float64
 5   College Green, Bank Of Ireland                   8783 non-null   float64
 6   Dame Street (Counter Missing)                    5897 non-null   float64
 7   Date & Time                                      8784 non-null   object 
 8   Dawson Street Replacement                        8783 non-null   float64
 9   Doilier Street, Burgh Quay    

In [23]:
data_2021 = pd.read_csv('raw_data/dcc-2021-pedestrian-footfall-31072021.csv')

In [90]:
data_2021 = data_2021.reindex(sorted(data_2021.columns), axis=1)

In [93]:
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5088 entries, 0 to 5087
Data columns (total 35 columns):
 #   Column                                           Non-Null Count  Dtype  
---  ------                                           --------------  -----  
 0   Aston Quay/Fitzgeralds                           2796 non-null   float64
 1   Bachelors walk/Bachelors way                     5087 non-null   float64
 2   Baggot st lower/Wilton tce inbound               5079 non-null   float64
 3   Baggot st upper/Mespil rd/Bank                   5079 non-null   float64
 4   Capel st/Mary street                             816 non-null    float64
 5   College Green/Bank Of Ireland                    5060 non-null   float64
 6   College Green/Church Lane                        5087 non-null   float64
 7   College st/Westmoreland st                       5087 non-null   float64
 8   D'olier st/Burgh Quay                            5087 non-null   float64
 9   Dame Street/Londis            

# Liste des capteurs ajoutés en 2021

2   Baggot st lower/Wilton tce inbound               5079 non-null   float64
 3   Baggot st upper/Mespil rd/Bank 

 15  Grand Canal st upp/Clanwilliam place             5078 non-null   float64
 16  Grand Canal st upp/Clanwilliam place/Google

Newcomen Bridge/Charleville mall inbound         5075 non-null   float64
 21  Newcomen Bridge/Charleville mall outbound        5074 non-null   float64
 22  North Wall Quay/Samuel Beckett bridge East       5087 non-null   float64
 23  North Wall Quay/Samuel Beckett bridge Wes

27  Phibsborough Rd/Enniskerry Road                  5087 non-null   float64
 28  Phibsborough Rd/Munster St                       798 non-null    float64
 29  Richmond st south/Portabello Harbour inbound     5079 non-null   float64
 30  Richmond st south/Portabello Harbour outbound 